src/main/webapp/app/exam/participate/summary/points-summary/grade-key-overview/grade-key-overview.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { GradingSystemService } from 'app/grading-system/grading-system.service';
+import { GradeStep, GradeStepsDTO } from 'app/entities/grade-step.model';
+import { HttpResponse } from '@angular/common/http';
+import { of } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { GradeType } from 'app/entities/grading-scale.model';
+
+@Component({
+    selector: 'jhi-grade-key-overview',
+    templateUrl: './grade-key-overview.component.html',
+    styleUrls: ['./grade-key-overview.scss'],
+})
+export class GradeKeyOverviewComponent implements OnInit {
+    constructor(private route: ActivatedRoute, private router: Router, private gradingSystemService: GradingSystemService) {}
+
+    courseId?: number;
+    examId?: number;
+
+    examTitle?: string;
+    gradeSteps: GradeStep[] = [];
+    studentGrade?: string;
+    isBonus = false;
+
+    ngOnInit(): void {
+        this.route.params.subscribe((params) => {
+            this.courseId = Number(params['courseId']);
+            this.examId = Number(params['examId']);
+            this.gradingSystemService
+                .findGradeStepsForExam(this.courseId, this.examId)
+                .pipe(catchError(() => of(new HttpResponse<GradeStepsDTO>({ body: { examTitle: '', gradeType: GradeType.GRADE, gradeSteps: [] }, status: 404 }))))
```

Comment:
I am wondering if this is the best way to handle the error, especially given that all fields are optional already.

Reply:
I handled it like that since this would display an empty table if no grade key is present, or if some error occurs for some reason. Do you have suggestions on how this could be improved?